src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AsType.cs

```diff
@@ -112,6 +113,26 @@ protected override bool RequiresPagedDataForParamCore(TexlNode[] args, int param
             // For the second argument, we need only metadata. No actual data from datasource is required.
             return paramIndex != 1;
         }
+
+        public override ICallNodeDelegatableNodeValidationStrategy GetCallNodeDelegationStrategy()
+        {
+            return base.GetCallNodeDelegationStrategy();
+        }
+    }
+
+    internal class AsTypeCallNodeDelegationStrategy : DelegationValidationStrategy
+    {
+        public AsTypeCallNodeDelegationStrategy(TexlFunction function)
+            : base(function)
+        {
+        }
+
+        public override bool IsValidCallNode(CallNode node, TexlBinding binding, OperationCapabilityMetadata metadata)
+        {
+            // AsType CallNodeDelegationStrategy should match default IsValidCallNode from
+            // DelegationValidationStrategy but skips only the !IsBlockScopedConstant check.
+            return IsValidCallNodeInternal(node, binding, metadata, true);
```

Comment:
>IsValidCallNodeInternal [](http://example.com/codeflow?start=19&length=23)

I don't think this is correct. You want to skip IsValidAsyncOrImpureNode check for As function.